storage/spanstore/composite_test.go

```diff
@@ -40,16 +39,16 @@ func (n *noopWriteSpanStore) WriteSpan(span *model.Span) error {
 }
 
 func TestCompositeWriteSpanStoreSuccess(t *testing.T) {
-	c := NewCompositeWriter(&noopWriteSpanStore{}, &noopWriteSpanStore{})
+	c := NewSequentialWriter(&noopWriteSpanStore{}, &noopWriteSpanStore{})
 	assert.NoError(t, c.WriteSpan(nil))
 }
 
 func TestCompositeWriteSpanStoreSecondFailure(t *testing.T) {
-	c := NewCompositeWriter(&errProneWriteSpanStore{}, &errProneWriteSpanStore{})
-	assert.EqualError(t, c.WriteSpan(nil), fmt.Sprintf("[%s, %s]", errIWillAlwaysFail, errIWillAlwaysFail))
+	c := NewSequentialWriter(&errProneWriteSpanStore{}, &errProneWriteSpanStore{})
+	assert.EqualError(t, c.WriteSpan(nil), errIWillAlwaysFail.Error())
```

Comment:
I don't think you can correctly test this way because you can't tell the difference between which writer returns the error. I suggest to change noop writer to take an optional err parameter that it always returns, and do a data driven test where you pass different permutations of 3 noop writers with and without errors and check for the right error in the end.